### PR TITLE
feat: new launchpad UI

### PR DESCRIPTION
**Pull request by elevatebart**
- Closes UNIFY-39 UNIFY-36 UNIFY-37 UNIFY-38

## Upgrading to ViteJs

In this PR I changed the launchpad build process to faster ViteJs.
I still have a bug when using `vite build --watch` which deletes the css after the first refresh. This is probably a bug in the windicss plugin.

## Upgrade the design

Screen Shot 2021-07-01 at 10 18 44 PM
Screen Shot 2021-07-01 at 10 19 13 PM
Screen Shot 2021-07-01 at 10 19 26 PM
Screen Shot 2021-07-01 at 10 19 44 PM

+ Setup iconify
+ add cypress component testing
+ customize some setups in windicss

### CODE REVIEW

File: packages/launchpad/src/store/app.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+interface State {
+  projectTitle: string
+  title: string
+  description: string
+  nextAction: () => void
+  backAction: () => void
+  alternativeAction?: () => void
+  steps: {
+      component?: boolean
```
Comment (lmiller1990) [inline packages/launchpad/src/store/app.ts:11]: Should these be optional? Maybe just `true` and `false`? I think we should only make things optional when `undefined` has a different meaning to `false`.

File: packages/launchpad/src/store/app.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      e2e?: boolean
+  }
+}
+
+function createInitialState (): State {
+  return {
+    projectTitle: 'design-system',
+    title: 'LaunchPad',
+    description: 'Scaffold Cypress Tests',
+    nextAction () {},
```
Comment (lmiller1990) [inline packages/launchpad/src/store/app.ts:25]: Nit: If we want to be defensive could do

```js
nextAction () {
  throw Error('Must assign an action before calling nextAction')
}
```

File: packages/launchpad/src/store/app.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    backAction () {},
+    steps: {},
+  }
+}
+
+const storeKey = Symbol('storeApp')
+
+export class StoreApp {
+  private readonly state: State;
```
Comment (lmiller1990) [inline packages/launchpad/src/store/app.ts:34]: We need linting, stay `;` everywhere

File: packages/launchpad/src/store/app.ts [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  getState () {
+    return readonly(this.state)
+  }
+
+  setMeta (meta: { title: string, description: string }) {
+    this.state.title = meta.title
+    this.state.description = meta.description
+  }
+
+  flagComponent (flag = true) {
```
Comment (lmiller1990) [inline packages/launchpad/src/store/app.ts:53]: Do we need a default value? I think we should require a value to be passed explicitly.

File: packages/launchpad/src/store/app.ts [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    this.state.backAction = newBack
+  }
+
+  onAlt (newAlt: () => void) {
+    this.state.alternativeAction = newAlt
+  }
+}
+
+// useful for testing
+export function createStoreApp (stateOverrides: Partial<State> = {}) {
```
Comment (lmiller1990) [inline packages/launchpad/src/store/app.ts:83]: nice!

File: packages/launchpad/src/store/config.ts [hunk 6]
```diff
@@ -0,0 +1,72 @@
+import { reactive, readonly, inject, App } from 'vue'
+import { Bundler } from '../utils/bundler'
+import { Framework } from '../utils/frameworks'
+import { TestingType } from '../utils/testingTypes'
+import { StoreApp } from './app'
+
+type ComponentSetup = {
```
Comment (lmiller1990) [inline packages/launchpad/src/store/config.ts:7]: I think we have a convention to use `interface` over `type` for object definitions (like below in `StateConfig`)

File: packages/launchpad/src/utils/bundler.ts [hunk 7]
```diff
@@ -0,0 +1,30 @@
+import LogoWebpack from '../images/logos/webpack.svg'
+import LogoVite from '../images/logos/vite.svg'
+
+export type BundlerId = 'webpack' | 'vite';
+
+export type Bundler = {
```
Comment (lmiller1990) [inline packages/launchpad/src/utils/bundler.ts:6]: `interface`? 🤔

File: packages/launchpad/src/utils/libraries.ts [hunk 8]
```diff
@@ -0,0 +1,17 @@
+export type Library = {
```
Comment (lmiller1990) [inline packages/launchpad/src/utils/libraries.ts:1]: `interface`

File: packages/launchpad/src/components/ButtonBar.spec.tsx [hunk 9]
```diff
@@ -0,0 +1,48 @@
+import ButtonBar from './ButtonBar.vue'
+
+describe('<ButtonBar />', () => {
+  it('playground', () => {
+    cy.mount(() => <ButtonBar next="Next Step" back="Back" />)
```
Comment (lmiller1990) [inline packages/launchpad/src/components/ButtonBar.spec.tsx:5]: Happy to use TSX here, kind of surprised it's not messing with React TSX (in terms of types).

Any benefit to TSX over 

```ts
mount(ButtonBar, {
    props: { next: 'Next Step' }
  }
})
```

?

**Review by lmiller1990 — APPROVED** [review]
Looks fine, left a few nitpicks.

Also, I think as we introduce GraphQL (assuming we do) a lot of the store stuff will be replaced by GraphQL - good to keep this in mind (the store structure might change).

**Comment by lmiller1990** [comment]
Looks beautiful, amazing work!

File: packages/launchpad/src/main.ts [hunk 10]
```diff
@@ -0,0 +1,17 @@
+import { createApp } from 'vue'
+import './main.scss'
+import 'virtual:windi.css'
+import App from './App.vue'
+import { createStoreApp } from './store/app'
+import { createStoreConfig } from './store/config'
+import ClickOutside from './plugins/ClickOutside'
```
[the remark this answers is not in the document]
Comment (elevatebart) [reply packages/launchpad/src/main.ts:7]: the "import". method is more explicit and more cypress-friendly.

File: packages/launchpad/src/components/ButtonBar.spec.tsx (hunk 9 above)
Comment (elevatebart) [reply packages/launchpad/src/components/ButtonBar.spec.tsx:5]: Easier to read ;)

**Pull request merged**